flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestBase.java

```diff
@@ -90,16 +90,27 @@
 public abstract class RestoreTestBase implements TableTestProgramRunner {
 
     private final Class<? extends ExecNode<?>> execNodeUnderTest;
+    private final List<Class<? extends ExecNode<?>>> childExecNodesUnderTest;
     private final AfterRestoreSource afterRestoreSource;
 
     protected RestoreTestBase(Class<? extends ExecNode<?>> execNodeUnderTest) {
         this.execNodeUnderTest = execNodeUnderTest;
+        this.childExecNodesUnderTest = new ArrayList<>();
+        this.afterRestoreSource = AfterRestoreSource.FINITE;
+    }
+
+    protected RestoreTestBase(
+            Class<? extends ExecNode<?>> execNodeUnderTest,
+            List<Class<? extends ExecNode<?>>> childExecNodesUnderTest) {
+        this.execNodeUnderTest = execNodeUnderTest;
+        this.childExecNodesUnderTest = childExecNodesUnderTest;
         this.afterRestoreSource = AfterRestoreSource.FINITE;
     }
```

Comment:
I strongly believe we should always redirect to a single ctor. Otherwise it's really hard to track what's initialised where.

I must've missed that when adding previous ctors.

Reply:
Would this be ok instead?
```
    protected RestoreTestBase(Class<? extends ExecNode<?>> execNodeUnderTest) {
        this(execNodeUnderTest, new ArrayList<>(), AfterRestoreSource.FINITE);
    }

    protected RestoreTestBase(
            Class<? extends ExecNode<?>> execNodeUnderTest,
            List<Class<? extends ExecNode<?>>> childExecNodesUnderTest) {
        this(execNodeUnderTest, childExecNodesUnderTest, AfterRestoreSource.FINITE);
    }

    protected RestoreTestBase(
            Class<? extends ExecNode<?>> execNodeUnderTest, AfterRestoreSource state) {
        this(execNodeUnderTest, new ArrayList<>(), state);
    }

    protected RestoreTestBase(
            Class<? extends ExecNode<?>> execNodeUnderTest,
            List<Class<? extends ExecNode<?>>> childExecNodesUnderTest,
            AfterRestoreSource state) {
        this.execNodeUnderTest = execNodeUnderTest;
        this.childExecNodesUnderTest = childExecNodesUnderTest;
        this.afterRestoreSource = state;
    }
```

Reply:
Yes, exactly! That's what I was hinting at 👍

Reply:
Fixed